rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -703,7 +788,14 @@ module RustDataFlow implements InputSig<Location> {
    * any value stored inside `f` is cleared at the pre-update node associated with `x`
    * in `x.f = newValue`.
    */
-  predicate clearsContent(Node n, ContentSet c) { none() }
+  predicate clearsContent(Node n, ContentSet c) {
+    exists(AssignmentExprCfgNode assignment, FieldExprCfgNode access |
+      assignment.getLhs() = access and
+      n.asExpr() = access.getExpr() and
+      access.getNameRef().getText().toInt() =
+        c.(SingletonContentSet).getContent().(TuplePositionContent).getPosition()
+    )
```

Comment:
This part should also give rise to a store step (but where the `toNode` is the post-update node of `n`). That should remove the false negative inside `tuple_mutation`.